### PR TITLE
enable spider attributes for the most common headers

### LINKED ISSUE #53: Spider attributes for the most common headers
Headers such as `X-Crawlera-Profile`, `X-Crawlera-Debug` and `X-Crawlera-Cookies` are quite often used by Crawlera users. As such it could be easier if this plugin offered builtin support for them as spider attributes, such as:

```python
class FooSpider(scrapy.Spider):
    name = 'foo'
    crawlera_profile = 'mobile'
    crawlera_debug = 'ua'
    ...
```

Without them, users always have to write their custom Crawlera middlewares or, even worse, pass the headers explicitly in each and every request issued by the spiders.

**A quite common use case:** you want to quickly debug some Crawlera settings such as user agents, response times, etc. To do that, you have to build a custom middleware for that or modify your spiders in every place where they generate requests. The attributes make it more convenient, as it'd be just a matter of setting `crawlera_debug = 'ua'`.

What do you think about adding support to such attributes in a spider level?

### PR DESCRIPTION
**Pull request by stummjr**
This PR fixes #53 by introducind support for Crawlera settings via spider attributes for some common headers, such as:

```python
class FooSpider(scrapy.Spider):
    name = 'foo'
    crawlera_profile = 'mobile'  # sets `X-Crawlera-Profile: mobile` header in all requests
    crawlera_debug = 'ua'  # sets `X-Crawlera-Debug: ua` header in all requests
    ...
```

I didn't add support to the full list of headers, as we are just starting the discussion here.

### CODE REVIEW

**Comment by eLRuLL** [comment]
@stummjr I actually like the idea. But few things to discuss:

- We'll have to give support for all crawlera available headers.
- What should have precedence here, `DEFAULT_REQUEST_HEADERS` or these ones?
- Should we also give settings-level support?

**Comment by baitxo** [comment]
@stummjr I also like the idea!

Regarding @eLRuLL questions:

- We'll have to give support for all crawlera available headers. 
Sure!

- What should have precedence here, DEFAULT_REQUEST_HEADERS or these ones?
I believe `DEFAULT_REQUEST_HEADERS` should have precedence. But we can raise a warning on `spider_opened` when we detect a collision. 

- Should we also give settings-level support?
I believe so! That way we could provide better defaults like it was proposed in #52

**Comment by eLRuLL** [comment]
after checking #52 I think it would be more interesting to have something like `DEFAULT_CRAWLERA_HEADERS`, which could actually help only define headers that are part of crawlera.

This doesn't conflict with `DEFAULT_REQUEST_HEADERS` as that will only be used if the `CrawleraMiddleware` is enabled.

We can also enable that on spider level with the argument of `default_crawlera_headers`.

So, sorry but I think the current proposal is not really that necessary. We even had this discussion a while ago and decided to only stay with `DEFAULT_CRAWLERA_HEADERS`, that's why we give emphasis on that on the [documentation](https://scrapy-crawlera.readthedocs.io/en/latest/).

**Comment by stummjr** [comment]
@eLRuLL @baitxo: I like the idea of having `DEFAULT_CRAWLERA_HEADERS` separated from `DEFAULT_REQUEST_HEADERS`. It is a very good replacement for the spider attributes I proposed here.

**Some pros:**

- no maintenance burden when new headers are added or old ones get deprecated;
- you have the same convenience at the spider level as with attributes, via `custom_settings`;
- no need to translate header names into proper attribute names and keep a mental map of that;
- avoid mistakenly sending `X-Crawlera-*` headers when the plugin isn't enabled (comparing with `DEFAULT_REQUEST_HEADERS`);


**tl;dr:** +1 to implement `DEFAULT_CRAWLERA_HEADERS`.
